e2e/testcases/permission_test.go

```diff
@@ -0,0 +1,387 @@
+			sourceChart:    privateCoreDNSHelmChart,
+			gsaEmail:       gsaRestrictedImpersonatedEmail(),
+			expectedErrMsg: missingHelmARPermissionError,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			nt := nomostest.New(t, nomostesting.WorkloadIdentity, ntopts.Unstructured, ntopts.RequireGKE(t))
+			rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
+			tenant := "tenant-a"
+
+			if err := workloadidentity.ValidateEnabled(nt); err != nil {
+				nt.T.Fatal(err)
+			}
+
+			mustConfigureMembership(nt, tc.fwiTest, false)
+
+			spec := &sourceSpec{
+				sourceType:    tc.sourceType,
+				sourceRepo:    tc.sourceRepo,
+				sourceChart:   tc.sourceChart,
+				sourceVersion: tc.sourceVersion,
+				gsaEmail:      tc.gsaEmail,
+			}
+			mustConfigureRootSync(nt, rs, tenant, spec)
+			nt.WaitForRootSyncSourceError(rs.Name, status.SourceErrorCode, tc.expectedErrMsg)
+		})
+	}
+
+}
+
+func TestKSAMissingReaderPermission(t *testing.T) {
+	testCases := []struct {
+		name           string
+		fwiTest        bool
+		sourceType     v1beta1.SourceType
+		AuthType       configsync.AuthType
+		sourceRepo     string
+		sourceChart    string
+		sourceVersion  string
+		gsaEmail       string
+		expectedErrMsg string
+	}{
+		{
+			name:           "WI-KSA - OCI source missing permission for GKE WI",
+			fwiTest:        false,
+			sourceType:     v1beta1.OciSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     privateARImage(),
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingOCIARPermissionError,
+		},
+		{
+			name:           "WI-KSA - Helm source missing permission for GKE WI",
+			fwiTest:        false,
+			sourceType:     v1beta1.HelmSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceVersion:  privateCoreDNSHelmChartVersion,
+			sourceChart:    privateCoreDNSHelmChart,
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingHelmARPermissionError,
+		},
+		{
+			name:           "WI-KSA - OCI source missing permission for Fleet WI",
+			fwiTest:        true,
+			sourceType:     v1beta1.OciSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceRepo:     privateARImage(),
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingOCIARPermissionError,
+		},
+		{
+			name:           "WI-KSA - Helm source missing permission for Fleet WI",
+			fwiTest:        true,
+			sourceType:     v1beta1.HelmSource,
+			AuthType:       configsync.AuthGCPServiceAccount,
+			sourceVersion:  privateCoreDNSHelmChartVersion,
+			sourceChart:    privateCoreDNSHelmChart,
+			gsaEmail:       gsaRestrictedEmail(),
+			expectedErrMsg: missingHelmARPermissionError,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			nt := nomostest.New(t, nomostesting.WorkloadIdentity,
+				ntopts.Unstructured,
+				ntopts.RequireGKE(t))
+
+			if err := workloadidentity.ValidateEnabled(nt); err != nil {
```

Comment:
Do we need WI enabled for KSA auth to work?

Reply:
This is just in our test setup though, right? 

Does the cluster actually need WI enabled in order to use BYOID? 

Any k8s cluster can inject a KSA into a Pod. 

Do we need to test BYOID on non-WI clusters?

Reply:
Nevermind. I misread your response.